### PR TITLE
12 add car browser frame

### PR DESCRIPTION
**Pull request by OthelloEngineer**

### CODE REVIEW

**Review by Ashvikan — APPROVED** [review]

File: src/components/molecules/CarCard.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+
+// Define the CarCard component to accept carName, carType, and imageLocation as props
+type CarCardProps = {
+    carName: string;
+    carType: string;
+    imageLocation: any;
+};
+
+export const CarCard = ({ carName, carType, imageLocation }: CarCardProps) => {
```
Comment (The0mikkel) [inline src/components/molecules/CarCard.tsx:14]: For components, I would love it if we could add some documentation on how they work. This would increase code quality.  
I will leave this as a comment, and it is more intended for future work.

File: src/components/molecules/CarCard.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    const [fontsLoaded] = useFonts({
+        Inter_500Medium,
+        Inter_700Bold,
+        Inter_300Light
+    });
```
Comment (The0mikkel) [inline src/components/molecules/CarCard.tsx:19]: Font loading have been moved to the `App.tsx` file, to decrease the number of times it has to be loaded.

**Review by Ashvikan — APPROVED** [review]

**Review by The0mikkel — CHANGES_REQUESTED** [review]
There are some *very* minor elements.  
There also seem to be a few formatting errors, but again, these are small.

I would love, however, to have fonts removed from the components, as it is handled on app level.

File: src/components/molecules/CarCard.tsx (hunk 1 above)
Comment (OthelloEngineer) [reply src/components/molecules/CarCard.tsx]: Thank you. I have added docs on its parameters and return types.

File: src/components/molecules/CarCard.tsx (hunk 2 above)
Comment (OthelloEngineer) [reply src/components/molecules/CarCard.tsx]: Awesome. 
I have removed the loading of the fonts in the component.

File: src/screens/CarBrowser.tsx [hunk 3]
```diff
@@ -1,12 +1,34 @@
...
 import { StatusBar } from 'expo-status-bar';
-import { StyleSheet, View, Button } from 'react-native';
-import { Text } from '../components/atoms/Text/Text';
+import { CarCard } from "../components/molecules/CarCard";
+import { View, StyleSheet, Text, TouchableOpacity, ScrollView } from "react-native";
+import { DefaultGradient } from "../components/molecules/DefaultGradient";
+import { Ionicons } from '@expo/vector-icons';
+import { CarsContext } from "../components/molecules/CarsContext";
 
 export function CarBrowser({ navigation }: { navigation: any }) {
```
Comment (The0mikkel) [inline src/screens/CarBrowser.tsx:9]: I sadly forgot a point, but this parameter list should be updated with the new format. 

The page should be setup in `App.tsx` like Home screen (I believe it already is though), as well as having the parameterlist updated. Please see Home screen on how to do this.

The listed cars should also be filtered by the location choosen on the Home screen (can be accessed through navigation route param list)

**Review by The0mikkel — CHANGES_REQUESTED** [review]

**Review by The0mikkel — CHANGES_REQUESTED** [review]
Please add to the discussion before re-requesting a review, by either updating the code or replying to my change request @OthelloEngineer

File: src/screens/CarBrowser.tsx (hunk 3 above)
Comment (OthelloEngineer) [reply src/screens/CarBrowser.tsx]: Is fixed with location filtering

File: package.json
Comment (jakob753951) [inline package.json:1]: this looks like irrelevant reordering

File: src/screens/CarBrowser.tsx [hunk 4]
```diff
@@ -1,23 +1,134 @@
...
+import { View, StyleSheet, Text, TouchableOpacity, ScrollView } from "react-native";
+import { DefaultGradient } from "../components/molecules/DefaultGradient";
+import { Ionicons } from '@expo/vector-icons';
+import { CarsContext } from "../components/molecules/CarsContext";
+import { NativeStackScreenProps } from "@react-navigation/native-stack";
+import { AppStackList } from "./App";
+import { Car } from "../utils/types/Car";
 
-export function CarBrowser({ navigation }: { navigation: any }) {
+type CarBrowserProps = NativeStackScreenProps<AppStackList, 'CarBrowser'>;
```
Comment (The0mikkel) [inline src/screens/CarBrowser.tsx:12]: There are some documentation missing

**Review by The0mikkel — CHANGES_REQUESTED** [review]

**Review by jakob753951 — APPROVED** [review]
maybe fix package.json, but other than that, good stuff :)

**Review by The0mikkel — APPROVED** [review]
